pipeline/terraform/stack/cluster.tf

```diff
@@ -1,3 +1,26 @@
 resource "aws_ecs_cluster" "cluster" {
-  name = local.namespace_hyphen
+  name               = local.namespace_hyphen
+  capacity_providers = [module.inference_capacity_provider.name]
+}
+
+module "inference_capacity_provider" {
+  source = "git::github.com/wellcomecollection/terraform-aws-ecs-service.git//modules/ec2_capacity_provider?ref=v3.1.0"
+
+  name = "${local.namespace_hyphen}_inference_capacity_provider"
+
+  // Setting this variable from aws_ecs_cluster.cluster.name creates a cycle
+  // The cluster name is required for the instance user data script
+  // This is a known issue https://github.com/terraform-providers/terraform-provider-aws/issues/12739
+  cluster_name = local.namespace_hyphen
+
+  instance_type           = "c5.2xlarge"
```

Comment:
very premium (@ 34c an hour) - but fine if this spends most of the time scaled to 0.

Reply:
The price should work out slightly less than what Fargate was costing us - this is also a spot instance

Reply:
My workings:
- Maxed out Fargate was $0.197 / hr, we ran <= 8 of them
- At spot pricing a `c5.2xlarge` is *maximum* $0.34, but actually has been consistently between $0.15-0.20 (over the last 3 months). We run <= 4 of them.
- I chose a 2xlarge (big enough for 2 tasks) as a compromise: for full reindexes we want maximum horizontal scaling to maximise throughput (so we fit 2 tasks on each host, 8 tasks max), in general use we want maximum speed at low-ish throughput so we give the 1 or 2 tasks we expect to see plenty of POWER. Anecdotally it seemed to run 1.5x-2x as fast as Fargate as I believe Pytorch is able to exploit certain features of the c5 hardware.

Reply:
More important than that logic is that I intend to keep an eye on utilisation and throughput and tweak these values where necessary!